src/renderer/components/tooltip/tooltip.tsx

```diff
@@ -112,26 +112,21 @@ export class Tooltip extends React.Component<TooltipProps> {
 
   @boundMethod
   refreshPosition() {
-    const { preferredPositions } = this.props;
+    const { preferredPositions = [] } = this.props;
     const { elem, targetElem } = this;
 
-    let positions = new Set<TooltipPosition>([
-      TooltipPosition.RIGHT,
-      TooltipPosition.BOTTOM,
-      TooltipPosition.TOP,
-      TooltipPosition.LEFT,
-      TooltipPosition.TOP_RIGHT,
-      TooltipPosition.TOP_LEFT,
-      TooltipPosition.BOTTOM_RIGHT,
-      TooltipPosition.BOTTOM_LEFT,
-    ]);
-
-    if (preferredPositions) {
-      positions = new Set([
-        ...[preferredPositions].flat(),
-        ...positions,
-      ]);
-    }
+    // Start with the preferred positions
+    const positions = new Set([preferredPositions].flat());
+
+    // Then add the default ordering, these won't override the above
+    positions.add(TooltipPosition.RIGHT);
+    positions.add(TooltipPosition.BOTTOM);
+    positions.add(TooltipPosition.TOP);
+    positions.add(TooltipPosition.LEFT);
+    positions.add(TooltipPosition.TOP_RIGHT);
+    positions.add(TooltipPosition.TOP_LEFT);
+    positions.add(TooltipPosition.BOTTOM_RIGHT);
+    positions.add(TooltipPosition.BOTTOM_LEFT);
```

Comment:
So this kind of refactoring looks like an improvement but makes your work take longer, the review take longer, testing take longer, impacts more code than this PR is intended for, and could introduce new bugs...

Reply:
Yes you are right, I shouldn't have committed that, will revert.